glean-core/csharp/Glean/Dispatchers.cs

```diff
@@ -130,140 +114,94 @@ internal static Task LaunchAPI(Action action)
                             {
                                 //TODO Exception eaten by Glean
                             }
-                        },
-                        cancellationToken,
-                        TaskCreationOptions.None,
-                        taskSchedulerPair.ExclusiveScheduler);
-                }
-                else
-                {
-                    // If we are in testing mode, then go ahead and invoke the
-                    // action to ensure synchronous execution.
-                    action.Invoke();
+                        });
+                    }
+                    else
+                    {
+                        // If we are in testing mode, then go ahead and await
+                        // the task to ensure synchronous execution.
+                        factory.StartNew(action).Wait();
+                    }
                 }
             }
 
             return task;
         }
 
         /// <summary>
-        /// This function launches an Action as an asynchronous Task on a
-        /// concurrently executed queue.
-        ///
-        /// **Note:** Tasks that need to be executed asynchronously but should
-        /// not block other tasks such as recording data should use the
-        /// `LaunchConcurrent` function. For example, tasks performed during
-        /// initialization or an upload task could be executed concurrently.
-        /// For tasks that need to be executed serially, see `LaunchAPI`.
-        ///
-        /// This function specifically ignores the `QueueInitialTasks` flag
-        /// because the only tasks that should be launched by this are the ping
-        /// upload schedulers and those should run regardless of the initialized
-        /// state.
-        ///
-        /// If `TestingMode` is enabled, then `LaunchConcurrent` will just
-        /// execute the task rather than adding it to the concurrent queue to
-        /// avoid asynchronous issues while testing.
+        /// Stops queueing Actions and processes any Actions in the queue. 
         /// </summary>
-        /// <param name="action">The Action to perform</param>
-        /// <returns>A Task or null if run synchronously</returns>
-        internal static Task LaunchConcurrent(Action action)
+        internal static void FlushQueuedInitialTasks()
         {
-            Task task = null;
+            // Create a list to store the created tasks in order to await
+            // their execution
+            List<Task> tasks = new List<Task>();
```

Comment:
Let's stick to the implementation on the Kotlin side: let's create a new action, that runs on the API scheduler, which is responsible for switching.